### PR TITLE
Change how we handle versioning

### PR DESCRIPTION
**Pull request by weshaggard**
- Eliminate the need for various queue-time variables
- The source of truth for the version is in the project
- For dev builds we override the prerelease version with dev-<date-based-bulid-number>

cc @mitchdenny @chidozieononiwu 

Note that the projects still need to be updated from `VersionPrefix` to `Version` and include the preview prerelease identifiers but @mitchdenny is working on that with other changes. In the mean time it shouldn't cause any issues people will just not have a `-preview.x` if they try to do a release.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (chidozieononiwu) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Would the $versioningProperties variable persist from the previous PowerShell step?

**Review by chidozieononiwu — APPROVED** [review]

**Review by mitchdenny — APPROVED** [review]

File: Documentation/Versioning.md
Comment (AlexGhiondea) [inline Documentation/Versioning.md]: This should only happen in the hotfix branch, right?

Is there a way to ensure that we don't merge this change back when the hotfix branch is merged back to master?

Comment (heaths) [inline Documentation/Versioning.md]: Do we plan - or allow for - shipping preview releases for hotfixes, like direct-to-customer private patches for them to try? If so, seems this should be called out similar to point 2 directly above.

File: eng/Versioning.targets
Comment (heaths) [inline eng/Versioning.targets]: Shouldn't the ".1" be the build revision number, though I'm not sure if that's available during build. In the legacy release pipelines, it was `$(Rev:r)` or `$(Rev:rr)` for double digits (good, if you expect that volume, since according to semver any prerelease metadata is lexigraphically sorted). That said, pipeline templates can define and use a [counter](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#counter) if `$(Rev)` is not available.

**Review by heaths — APPROVED** [review]

File: Documentation/Versioning.md
Comment (weshaggard) [reply Documentation/Versioning.md]: I don't think that will be a general flow but certainly possible, if needed.

File: eng/Versioning.targets
Comment (weshaggard) [reply eng/Versioning.targets]: That is what is coming into OfficialBuildId property but for local dev builds I just hard-code that to 1.

File: Documentation/Versioning.md
Comment (weshaggard) [reply Documentation/Versioning.md]: Generally it will only be in a hotfix branch but that isn't required. There isn't any great way to prevent it from merging into master, people just need to correctly resolve the merge conflicts and not pull the change in.

**Review by AlexGhiondea — APPROVED** [review]

**Pull request merged**
